src/Auth/GrantQueryBuilderInterface.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\TableBackendUtils\Auth;
+
+use Keboola\TableBackendUtils\Auth\Grant\GrantOptionsInterface;
+use Keboola\TableBackendUtils\Auth\Grant\RevokeOptionsInterface;
+
+interface GrantQueryBuilderInterface
+{
+    /**
+     * @param string[] $permissions
+     * @param string[] $grantOnTargetPath
+     */
+    public function getGrantSql(
+        array $permissions,
+        ?string $grantSubject,
+        array $grantOnTargetPath,
+        string $to,
+        ?GrantOptionsInterface $options
+    ): string;
+
+    /**
+     * @param string[] $permissions
+     * @param string[] $grantOnTargetPath
+     */
+    public function getRevokeSql(
+        array $permissions,
+        ?string $grantSubject,
+        array $grantOnTargetPath,
+        string $to,
+        ?RevokeOptionsInterface $options
+    ): string;
```

Comment:
Já si myslím že na tohle by měl vzniknout Options objekt. Už to má dost parametrů, navíc některé nepovinné.

Reply:
Mno nevím co je lepší tak tu konstrukci jen přesuneš do jiného objektu. Takto je to podobné syntaxe toho sql grantu.

Reply:
Opravil sem to na option objekty s fluent interface, není to moc podle mého gusta.

Reply:
Super takhle jsem to zamýšlel. Kouknu na to